applications/GeoMechanicsApplication/custom_elements/small_strain_U_Pw_diff_order_element.cpp

```diff
@@ -189,94 +189,86 @@ void SmallStrainUPwDiffOrderElement::Initialize(const ProcessInfo& rCurrentProce
 {
     KRATOS_TRY
 
-    const GeometryType&                             rGeom = GetGeometry();
-    const GeometryType::IntegrationPointsArrayType& IntegrationPoints =
-        rGeom.IntegrationPoints(this->GetIntegrationMethod());
-
-    if (mConstitutiveLawVector.size() != IntegrationPoints.size())
-        mConstitutiveLawVector.resize(IntegrationPoints.size());
+    const auto& r_properties = GetProperties();
+    const auto& r_geometry   = GetGeometry();
+    const auto number_of_integration_points = r_geometry.IntegrationPointsNumber(GetIntegrationMethod());
 
-    if (GetProperties()[CONSTITUTIVE_LAW] != nullptr) {
-        for (unsigned int i = 0; i < mConstitutiveLawVector.size(); ++i) {
-            mConstitutiveLawVector[i] = GetProperties()[CONSTITUTIVE_LAW]->Clone();
-            mConstitutiveLawVector[i]->InitializeMaterial(
-                GetProperties(), rGeom, row(rGeom.ShapeFunctionsValues(this->GetIntegrationMethod()), i));
-        }
-    } else
-        KRATOS_ERROR << "A constitutive law needs to be specified for the "
-                        "element with ID "
-                     << this->Id() << std::endl;
+    mConstitutiveLawVector.resize(number_of_integration_points);
 
-    // Retention law
-    if (mRetentionLawVector.size() != IntegrationPoints.size())
-        mRetentionLawVector.resize(IntegrationPoints.size());
+    for (unsigned int i = 0; i < mConstitutiveLawVector.size(); ++i) {
+        mConstitutiveLawVector[i] = r_properties[CONSTITUTIVE_LAW]->Clone();
+        mConstitutiveLawVector[i]->InitializeMaterial(
+            r_properties, r_geometry, row(r_geometry.ShapeFunctionsValues(GetIntegrationMethod()), i));
+    }
 
+    mRetentionLawVector.resize(number_of_integration_points);
     for (unsigned int i = 0; i < mRetentionLawVector.size(); ++i) {
-        mRetentionLawVector[i] = RetentionLawFactory::Clone(GetProperties());
+        mRetentionLawVector[i] = RetentionLawFactory::Clone(r_properties);
         mRetentionLawVector[i]->InitializeMaterial(
-            GetProperties(), rGeom, row(rGeom.ShapeFunctionsValues(this->GetIntegrationMethod()), i));
+            r_properties, r_geometry, row(r_geometry.ShapeFunctionsValues(GetIntegrationMethod()), i));
     }
 
-    const SizeType NumUNodes = rGeom.PointsNumber();
-    const SizeType NumDim    = rGeom.WorkingSpaceDimension();
+    if (mStressVector.size() != number_of_integration_points) {
+        mStressVector.resize(number_of_integration_points);
+        for (unsigned int i = 0; i < mStressVector.size(); ++i) {
+            mStressVector[i].resize(GetStressStatePolicy().GetVoigtSize());
+            std::fill(mStressVector[i].begin(), mStressVector[i].end(), 0.0);
+        }
+    }
```

Comment:
If the number of integration points is the same, but the VoigtSize is different, the elements will keep a wrong size?

Reply:
I believe that this part of the initialization will always be done, and therefore a size check may indeed be redundant. However, I wouldn't want to change the behavior of this function yet. So I would suggest to change this in a separate PR, if you agree.